crates/ra_syntax/src/syntax_error.rs

```diff
@@ -84,6 +84,9 @@ pub enum SyntaxErrorKind {
     ParseError(ParseError),
     EscapeError(EscapeError),
     TokenizeError(TokenizeError),
+    // FIXME: the obvious pattern of this enum dictates that the following enum variants
+    // should be wrapped into something like `SemmanticError(SemmanticError)`
+    // or `ValidateError(ValidateError)` or `SemmanticValidateError(...)`
```

Comment:
Okay, let's leave it for the refactoring to be done soon. Since this crate is called `ra_syntax` let's leave `SyntaxError` name for that and not some other `OpaqueError`.
If this crate is used only be rust-analyzer exclusively I am definitely for generic `(String, TextRange)` error shape.